aws_generated_data/commands/rds_eol.py

```diff
@@ -0,0 +1,150 @@
+import logging
+from collections.abc import Iterable
+from datetime import (
+    date,
+    datetime,
+    timedelta,
+)
+from pathlib import Path
+from typing import Any
+
+import requests
+import typer
+import yaml
+from bs4 import BeautifulSoup
+from pydantic import (
+    BaseModel,
+    RootModel,
+    ValidationError,
+)
+from typing_extensions import Annotated
+
+app = typer.Typer()
+log = logging.getLogger(__name__)
+
+
+class RdsItem(BaseModel):
+    engine: str
+    version: str
+    eol: date
+
+    def __hash__(self) -> int:
+        return hash((self.engine, self.version))
+
+    def __lt__(self, other: Any) -> bool:
+        if not isinstance(other, RdsItem):
+            return False
+        return (self.engine, self.version) < (other.engine, other.version)
+
+
+RdsEol = RootModel[list[RdsItem]]
+CalItem = tuple[str, datetime]
+
+
+class Engine:
+    def __init__(self, value: str):
+        self.name, self.url = value.split(":", maxsplit=1)
+
+    def __str__(self) -> str:
+        return f"<Engine: {self.name=} {self.url=}>"
+
+    def __repr__(self) -> str:
+        return str(self)
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Engine):
+            return False
+        return (self.name, self.url) == (other.name, other.url)
+
+
+def engine_with_url(value: str) -> Engine:
+    return Engine(value)
+
+
+def parse_date(date_str: str) -> datetime:
+    try:
+        return datetime.strptime(date_str, "%B %Y")
+    except ValueError:
+        return datetime.strptime(date_str, "%d %B %Y")
+
+
+def parse_aws_release_calendar(page: str) -> list[CalItem]:
+    items: list[CalItem] = []
+    soup = BeautifulSoup(page, "html.parser")
+    version_table = soup.find_all("table")[0]
```

Comment:
Find first `table` seems a little hard coded, but I can't predicate how that page may change in the future, so good for now.

Btw, to find first, can use `soup.find("table")` directly.

Reply:
Changed in [882c573](https://github.com/app-sre/aws-generated-data/pull/1/commits/882c573a52150cf134b79a12043780309542da21)